core/src/main/scala/cats/instances/stream.scala

```diff
@@ -118,9 +118,6 @@ trait StreamInstances extends cats.kernel.instances.StreamInstances {
 
       override def collect[A, B](fa: Stream[A])(f: PartialFunction[A, B]): Stream[B] = fa.collect(f)
 
-      override def foldM[G[_], A, B](fa: Stream[A], z: B)(f: (B, A) => G[B])(implicit G: Monad[G]): G[B] =
-        Foldable.iteratorFoldM(fa.toIterator, z)(f)
```

Comment:
Patten match Stream into `a #:: tail` will force an evaluation of the head of the `tail`, so I used vanilla `if`